### ISSUE TITLE
spack-stack-1.3.0 release work

### ISSUE THREAD

**Issue by climbfuji** [JCSDA/spack-stack#503]
## Description

This epic encompasses all the work require for the spack-stack-1.3.0 release.

- [x] Create release branches `release/1.3.0` in spack-stack and spack, point spack-stack release branch to spack release branch
- [x] Update container recipes, including adding mysql server+client: https://github.com/NOAA-EMC/spack-stack/pull/505
- [x] Update documentation for release (general updates and site config updates - install locations etc):
    - [x] #526 for all configs except NOAA Parallel Works, Acorn; one open question for Gaea to resolve (all in `PreConfiguredSites.rst`)
    - [x] https://github.com/NOAA-EMC/spack-stack/pull/536
- [x]  Install on all supported tier-1 platforms with latest Intel compiler and GNU compiler (in one unified environment using the compiler matrix) and test with JEDI-Skylab, ufs-weather-model, ufs-srweather-app, global workflow (except GSI). Skip Intel 18. Note. Default versions for unified-dev environment are: `esmf@8.3.0b09`, `mapl@2.22.0`, `crtm@2.4.0` and `crtm@v2.4-jedi.2`. Additional versions are installed manually afterwards
    - [x] Acorn: `/lfs/h1/emc/nceplibs/noscrub/spack-stack/spack-stack-1.3.0/envs/unified-env`(@AlexanderRichert-NOAA)
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Casper: `/glade/work/epicufsrt/contrib/spack-stack/spack-stack-1.3.0-casper/envs/unified-env` (@climbfuji)
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Cheyenne: `/glade/work/epicufsrt/contrib/spack-stack/spack-stack-1.3.0/envs/unified-env` (@ulmononian - rebuild @climbfuji)
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Discover: ` /gpfsm/dswdev/jcsda/spack-stack/spack-stack-1.3.0/envs/unified-env` (@climbfuji)
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Gaea (C3/C4): `/lustre/f2/dev/wpo/role.epic/contrib/spack-stack/spack-stack-1.3.0/envs/unified-env` (@climbfuji)
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Hera: `/scratch1/NCEPDEV/nems/role.epic/spack-stack/spack-stack-1.3.0/envs/unified-env` (@mark-a-potts - rebuild @climbfuji)
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] **Note** Manual modification of installed ESMF/MAPL libraries to move shared libraries out of the way (no longer needed if #532 is used for the rebuild)
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Jet: `/mnt/lfs4/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.3.0/envs/unified-env` (@ulmononian)
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Narwhal Intel: /p/app/projects/NEPTUNE/spack-stack/spack-stack-1.3.0/envs/unified-env-intel-2021.4.0 (@climbfuji) - separate from GNU, see https://github.com/NOAA-EMC/spack-stack/issues/523
        - [x] Added `esmf@8.4.1` (but not `mapl@2.35.2` - see https://github.com/NOAA-EMC/spack-stack/issues/524) in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Narwhal GNU: /p/app/projects/NEPTUNE/spack-stack/spack-stack-1.3.0/envs/unified-env-gcc-10.3.0 (@climbfuji) - separate from Intel, see https://github.com/NOAA-EMC/spack-stack/issues/523
        - [x] Added `esmf@8.4.1` (but not `mapl@2.35.2` - see https://github.com/NOAA-EMC/spack-stack/issues/524) in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Orion: `/work/noaa/epic-ps/role-epic-ps/spack-stack/spack-stack-1.3.0/envs/unified-env` (@ulmononian - rebuild @climbfuji)
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] S4: `/data/prod/jedi/spack-stack/spack-stack-1.3.0/envs/unified-env` (@climbfuji)
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Extra credit: unified-env on @climbfuji's macOS: `/Users/heinzell/prod/spack-stack-1.3.0/envs/unified-env`
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Extra credit: skylab-env on @climbfuji's macOS: `/Users/heinzell/prod/spack-stack-1.3.0/envs/skylab-env`
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Extra credit: unified-env on AWS EC2 Red Hat 8 (single node) AMI: `/home/ec2-user/spack-stack/spack-stack-1.3.0/envs/unified-env` (@climbfuji) - available on request as [ami-04a74896e17b9f9a0](https://us-east-1.console.aws.amazon.com/ec2/v2/home?region=us-east-1#ImageDetails:imageId=ami-04a74896e17b9f9a0) in us-east-1 and [ami-0d4506dfd5cf2168f](https://us-east-2.console.aws.amazon.com/ec2/v2/home?region=us-east-2#ImageDetails:imageId=ami-0d4506dfd5cf2168f) in us-east-2
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
    - [x] Extra credit: unified-env on AWS Parallel Cluster Ubuntu 20.04 (JCSDA): `/mnt/experiments-efs/skylab-v4/spack-stack-1.3.0/envs/unified-env` (@climbfuji)
        - [x] Added `esmf@8.4.1`+`mapl@2.35.2` in release and debug mode
        - [x] Added `crtm@v3.0.0-rc.1`
        - [x] Verified correct module names for mapl modules
        - [x] Rebuild with @climbfuji's bug fixes (https://github.com/NOAA-EMC/spack-stack/pull/532)
- Skip for now, but install later from develop:
    - Gaea (C5): not officially supported in this release, system still not released for production  
    - Hercules: not officially supported in this release, system still not released for production  
    - NOAA Parallel Works AWS/Azure/GCP: `/contrib/EPIC/spack-stack/spack-stack-1.3.0/envs/unified-dev` (@mark-a-potts)
- [x]  Test with JEDI-Skylab, ufs-weather-model, ufs-srweather-app, global workflow (except GSI). Skip Intel 18.
    - [x] Hera:
        - [x] JEDI-UFS (`ufs-bundle` so far tested Intel only)
        - [x] ufs-weather-model
        - [x] ufs-srw
    - [x] Jet:
        - [x] ufs-srw
    - [x] Narwhal (GNU):
        - [x] jedi-skylab
    - [x] Orion:
        - [x] jedi-skylab
        - [x] ufs-srw
    - [x] S4: 
        - [x] jedi-skylab
    - [x] AWS Parallel Cluster Ubuntu 20.04 (JCSDA):
        - [x] JEDI-UFS (`ufs-bundle`; tested Intel only)
        - [x] jedi-skylab (`jedi-bundle`; tested Intel only)
- [x] Tag repositories and update release notes

**Comment by ulmononian** [comment]
Orion Intel/GNU installation initiated in `/work/noaa/epic-ps/role-epic-ps/spack-stack/spack-stack-1.3.0`.

**Comment by AlexanderRichert-NOAA** [comment]
@ulmononian Under `envs/unified-env/install/`, I don't see intel/, just gcc/, and I don't see modulefiles/.

**Comment by ulmononian** [comment]
> @ulmononian Under `envs/unified-env/install/`, I don't see intel/, just gcc/, and I don't see modulefiles/.

the installation just started, so i assume it will go through gcc first judging from the concretize log.

**Comment by AlexanderRichert-NOAA** [comment]
Oh sorry I missed the word "initiated" :) We're good...

Who's doing Hera? Unfortunately I don't think I have the right access to do the installation.

**Comment by climbfuji** [comment]
From slack: 

hera/jet - would be good if this was EMC or EPIC, but I can help if needed.

Hang is doing wcoss2/acorn.

We/I/JCSDA can take care of discover, s4.

We need to decide who does gaea (and gaea c5), hercules (the new add-on to orion), cheyenne, casper. Happy to take the lead on hercules, gaea c5 etc.

maybe a short chat?

**Comment by AlexanderRichert-NOAA** [comment]
Yeah if we can fill out the list of locations etc. together that'd be great.

**Comment by climbfuji** [comment]
I’ll send an invite for this afternoon

**Comment by climbfuji** [comment]
While it doesn't make a difference right now (because the branches are identical), please continue to install from release/1.3.0 branch on other platforms:
```
git clone -b release/1.3.0 --recursive https://github.com/noaa-emc/spack-stack spack-stack-1.3.0
```

**Comment by ulmononian** [comment]
> While it doesn't make a difference right now (because the branches are identical), please continue to install from release/1.3.0 branch on other platforms:
> 
> ```
> git clone -b release/1.3.0 --recursive https://github.com/noaa-emc/spack-stack spack-stack-1.3.0
> ```

10-4. this was what i did for orion.

**Comment by ulmononian** [comment]
Cheyenne UE install initiated here:  `/glade/work/epicufsrt/contrib/spack-stack/spack-stack-1.3.0/envs/unified-env`
Jet UE install initiated here: `/mnt/lfs4/HFIP/hfv3gfs/role.epic/spack-stack/spack-stack-1.3.0/envs/unified-env`

**Comment by climbfuji** [comment]
S4 install initiated here: `/data/prod/jedi/spack-stack/spack-stack-1.3.0/envs/unified-env`

**Comment by ulmononian** [comment]
orion srw tests successful: `/work2/noaa/epic-ps/cbook/ue_srw_testing/OUE/cbSRW_testUE/tests/fundamental.orion_tests`

**Comment by climbfuji** [comment]
> orion srw tests successful: `/work2/noaa/epic-ps/cbook/ue_srw_testing/OUE/cbSRW_testUE/tests/fundamental.orion_tests`

Great! I think it's time to expand the PR description above and add the testing boxes

**Comment by ulmononian** [comment]
jet UE rebuilt w/ core2 target; mapl, crtm, esfm add-on complete. 

all jet fundamental srw tests  completed successfully: `/lfs4/HFIP/hfv3gfs/Cameron.Book/sw/cbSRW-testUE/tests/WE2E/ue_intel`.

**Comment by AlexanderRichert-NOAA** [comment]
Acorn is finally done (/lfs/h1/emc/nceplibs/noscrub/spack-stack/spack-stack-1.3.0/envs/unified-env; just waiting on newer esmf/mapl to finish), with the following modifications/caveats:
- Switched to GCC 11.2.0 (due to Acorn-specific Python installation issues)
- Changed py-scipy version to avoid use of python meson (build was failing with resource unavailable errors in forked compilation processes for both compilers)
- Had to load 'PrgEnv-gnu gcc craype' to get ESMF to build. This is a scary one. ESMF's `setup_build_environment` can't find the fortran compiler (needed when compiling with GCC) unless I do this. So the shell environment is impacting the build environment. Bleh.
- Removed +curl from GDAL to let it build with Intel.
- Moved build stage directory to /tmp to avoid annoying failures due to LFS problems.
- Set umask 002 and that seemed to take care of the problem of wrong group ownership/setgid.

I'll push the updated site config shortly after I test with the static ufs stack. @climbfuji, any preference as to whether I push to 1.3.0 or develop? Acorn doesn't need any spack changes, just spack-stack.

**Comment by climbfuji** [comment]
> Acorn is finally done (/lfs/h1/emc/nceplibs/noscrub/spack-stack/spack-stack-1.3.0/envs/unified-env; just waiting on newer esmf/mapl to finish), with the following modifications/caveats:
> 
> * Switched to GCC 11.2.0 (due to Acorn-specific Python installation issues)
> * Changed py-scipy version to avoid use of python meson (build was failing with resource unavailable errors in forked compilation processes for both compilers)
> * Had to load 'PrgEnv-gnu gcc craype' to get ESMF to build. This is a scary one. ESMF's `setup_build_environment` can't find the fortran compiler (needed when compiling with GCC) unless I do this. So the shell environment is impacting the build environment. Bleh.
> * Removed +curl from GDAL to let it build with Intel.
> * Moved build stage directory to /tmp to avoid annoying failures due to LFS problems.
> * Set umask 002 and that seemed to take care of the problem of wrong group ownership/setgid.
> 
> I'll push the updated site config shortly after I test with the static ufs stack. @climbfuji, any preference as to whether I push to 1.3.0 or develop? Acorn doesn't need any spack changes, just spack-stack.

Thanks @AlexanderRichert-NOAA. If your changes are ready before we tag, then let's merge them into the release branch (assuming it only affects the acorn site config). If it affects others as well, then better to develop. I saw the same problem on Narwhal as you on Acordn for ESMF: the shell environment is impacting the build environment. Echoing the "Bleh"!
